synthetic_CST.py

```diff
@@ -114,285 +167,116 @@ def melt_conc(i):
+        time_2_cum_dist = new_fuel(temp_1.tail(1)['ts'].item(),tem_1.head(1)['ts'].item(),
+                               temp_1.tail(1)['cum_distance'].item(),
+                               tem_1.head(1)['cum_distance'].item(),pd.to_datetime(time_2))
+#     else:
+#         time_2_level = 0;time_2_dist=0
     time_3 = str(group.head(1)['date'].item())+' 22:00:00'
     temp_2 = cst_term[cst_term['ts']<pd.to_datetime(time_3)]
     tem_2 = cst_term[cst_term['ts']>pd.to_datetime(time_3)]
     if len(tem_2)==0:
         time_3_level = temp_2.tail(1)['currentFuelVolumeTank1'].item()
-        time_3_dist=temp_2.tail(1)['Distance'].item()
+#         time_3_dist=temp_2.tail(1)['Distance'].item()
+        time_3_cum_dist=temp_2.tail(1)['cum_distance'].item()
     elif len(temp_2)==0:
+#         print(group.head(1)['termid'].item(),group.head(1)['date'].item())
         time_3_level=tem_2.head(1)['currentFuelVolumeTank1'].item()
-        time_3_dist=tem_2.head(1)['Distance'].item()
+#         time_3_dist=tem_2.head(1)['Distance'].item()
+        time_3_cum_dist=tem_2.head(1)['cum_distance'].item()
     else:
         time_3_level = new_fuel(temp_2.tail(1)['ts'].item(),tem_2.head(1)['ts'].item(),
                                temp_2.tail(1)['currentFuelVolumeTank1'].item(),
                                tem_2.head(1)['currentFuelVolumeTank1'].item(),pd.to_datetime(time_3))
-        time_3_dist = new_fuel(temp_2.tail(1)['ts'].item(),tem_2.head(1)['ts'].item(),
-                               temp_2.tail(1)['Distance'].item(),
-                               tem_2.head(1)['Distance'].item(),pd.to_datetime(time_3))
+#         time_3_dist = new_fuel(temp_2.tail(1)['ts'].item(),tem_2.head(1)['ts'].item(),
+#                                temp_2.tail(1)['Distance'].item(),
+#                                tem_2.head(1)['Distance'].item(),pd.to_datetime(time_3))
+        time_3_cum_dist = new_fuel(temp_2.tail(1)['ts'].item(),tem_2.head(1)['ts'].item(),
+                               temp_2.tail(1)['cum_distance'].item(),
+                               tem_2.head(1)['cum_distance'].item(),pd.to_datetime(time_3))
     temp_df = pd.DataFrame({'ts':[time_1,time_2,time_3],'currentFuelVolumeTank1':[time_1_level,time_2_level,time_3_level],
-                           'Distance':[time_1_dist,time_2_dist,time_3_dist]})
+                           'cum_distance':[time_1_cum_dist,time_2_cum_dist,time_3_cum_dist]})
     temp_df['termid'] = group.head(1)['termid'].item()
     temp_df['regNumb'] = group.head(1)['regNumb'].item()
     temp_df['ts'] = pd.to_datetime(temp_df['ts'])
     temp_df['mine'] = group.head(1)['mine'].item();temp_df['class'] = group.head(1)['class'].item()
     df = pd.concat([group,temp_df],axis=0)
-    df.sort_values(by=['ts'],inplace=True)
+    df.sort_values(by=['ts','direction'],na_position='last',inplace=True)
+    df.drop_duplicates(subset=['ts'],inplace=True)
     return df
 
 def custom_function(group):
```

Comment:
@Manjimadutta @pronoym99 : function names should be self explanatory. Avoid general names like `custom_function`